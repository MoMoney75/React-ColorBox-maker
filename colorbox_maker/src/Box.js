const Box = ({backgroundColor,height,width, handleRemove}) =>{

    const remove = () => handleRemove()
    return(
        <div>
        <div style={{backgroundColor : backgroundColor,
                     width: width + "px", 
                    height: height + "px"}}
                    >
        
        </div>
        <button onClick={remove}>x</button>
        </div>
        
    )
}

export default Box;