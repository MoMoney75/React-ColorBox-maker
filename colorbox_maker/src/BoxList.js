import { useState } from "react";
import Box from "./Box";
import BoxForm from "./NewBoxForm";



const BoxList = () => {
const [boxes,setBoxes] = useState([])
const handleFormSubmit = (formData)=>{
    
setBoxes((prevBoxes) =>  [...prevBoxes,formData])
}
const remove = (index) =>{
    setBoxes(boxes.filter((box,i) => i !== index))
}


return(
    
  <div>
<BoxForm onFormSubmit={handleFormSubmit} />
<ul className="BoxList">
{boxes.map((box,index) => (
<li>
<Box 
key={index}
backgroundColor = {box.backgroundColor}
width = {box.width}
height={box.height}
handleRemove={() => remove(index)}

/>

</li>
))}
</ul>
  </div>
)

};
export default BoxList;