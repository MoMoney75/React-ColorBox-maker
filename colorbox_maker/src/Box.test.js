import React from "react"
import BoxList from "./BoxList"
import {render} from "@testing-library/react"
import '@testing-library/jest-dom/extend-expect'
import Box from "./Box"



test("Smoke Test: Renders main BoxList component", function(){
render(<BoxList/>)
});

test("Snapshot test of BoxList", function(){
    const {asFragment} = render(< BoxList />)
    expect(asFragment()).toMatchSnapshot()
})


test("Smoke Test: Renders individual Box component", function(){
    render(<Box/>)
    });

test("Snapshot test of Box", function(){
        const {asFragment} = render(< Box />)
        expect(asFragment()).toMatchSnapshot()
    })