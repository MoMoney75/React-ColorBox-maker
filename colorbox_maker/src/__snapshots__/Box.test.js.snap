// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of Box 1`] = `
<DocumentFragment>
  <div>
    <div />
    <button>
      x
    </button>
  </div>
</DocumentFragment>
`;

exports[`Snapshot test of BoxList 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h1>
        Create a new box!
      </h1>
      <form>
        <label
          for="backgroundColor"
        >
          background color:
        </label>
        <input
          id="backgroundColor"
          name="backgroundColor"
          type="text"
          value=""
        />
        <label
          for="width"
        >
           width:
        </label>
        <input
          id="width"
          name="width"
          type="text"
          value=""
        />
        <label
          for="height"
        >
          height:
        </label>
        <input
          id="height"
          name="height"
          type="text"
          value=""
        />
        <button>
          submit
        </button>
      </form>
    </div>
    <ul
      class="BoxList"
    />
  </div>
</DocumentFragment>
`;
